File: 2026-09/src/day_05/prms_120863/Solution.cpp
#include <bits/stdc++.h>
using namespace std;

string solution(string polynomial) {
    stringstream ss(polynomial);
    string token;
    vector<int> cnt(2);

    while (ss >> token) {
        if (token == "+") continue;

        if (token.ends_with("x")) {
            if (token.size() == 1) {
                cnt[0]++;
            } else {
                cnt[0] += stoi(token.substr(0, token.size() - 1));
            }
        } else {
            cnt[1] += stoi(token);
        }
    }

    if (cnt[0] == 0) {
        return to_string(cnt[1]);
    } else if (cnt[1] == 0) {
        return (cnt[0] == 1 ? "" : to_string(cnt[0])) + "x";
    } else {
        return (cnt[0] == 1 ? "" : to_string(cnt[0])) + "x + " + to_string(cnt[1]);
    }
}

File: 2026-09/src/day_05/prms_120863/Solution.py
def solution(polynomial):
    cnt = [0, 0]
    for token in polynomial.split(" + "):
        if token.endswith("x"):
            cnt[0] += int(token[:-1] or 1)
        else:
            cnt[1] += int(token)

    res = []
    if cnt[0]:
        res.append("x" if cnt[0] == 1 else f"{cnt[0]}x")
    if cnt[1]:
        res.append(str(cnt[1]))

    return " + ".join(res)
